lib/rdoc/yard.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+##
+# YARD compatibility module for RDoc
+#
+# This module provides support for parsing YARD tags in RDoc comments.
+# Currently supports:
+# - @yield [params] - Describes block parameters
+# - @private - Marks method/class as private
+# - @api private - Alternative private marking
+
+class RDoc::YARD
```

Comment:
Can we improve this name?

I think that "YARD" isn't suitable here because we don't support full YARD features. We only focus on YARD tags here.